File: app/src/main/java/com/channa/mobiledatausageapp/viewmodel/MobileDataUsageViewModel.java
package com.channa.mobiledatausageapp.viewmodel;

import com.channa.mobiledatausageapp.data.model.Year;
import com.channa.mobiledatausageapp.repository.MobileDataRepository;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class MobileDataUsageViewModel extends ViewModel {

    MobileDataRepository mobileDataRepository;

    public MobileDataUsageViewModel(MobileDataRepository mobileDataRepository) {
        this.mobileDataRepository = mobileDataRepository;
    }

    public LiveData<List<Year>> getYearlyMobileDataUsage() {
        return mobileDataRepository.getYearlyMobileDataUsage();
    }
}

File: app/src/main/java/com/zak/datauses/viewmodel/MobileDataUsageViewModel.kt
package com.zak.datauses.viewmodel

import com.zak.datauses.repository.MobileDataRepository
import androidx.lifecycle.ViewModel
import androidx.lifecycle.LiveData
import com.zak.datauses.data.model.YearListWrapper

class MobileDataUsageViewModel(var mobileDataRepository: MobileDataRepository) : ViewModel() {
    val yearlyMobileDataUsage: LiveData<YearListWrapper>
        get() = mobileDataRepository.yearlyMobileDataUsage
}